File: remote/src/main/java/com/kelsos/mbrc/ui/fragments/LyricsFragment.java
package com.kelsos.mbrc.ui.fragments;

import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import butterknife.Bind;
import butterknife.ButterKnife;
import com.google.inject.Inject;
import com.kelsos.mbrc.R;
import com.kelsos.mbrc.adapters.LyricsAdapter;
import com.kelsos.mbrc.events.ui.LyricsUpdated;
import com.squareup.otto.Bus;
import com.squareup.otto.Subscribe;
import java.util.ArrayList;
import java.util.Arrays;
import roboguice.fragment.RoboFragment;

public class LyricsFragment extends RoboFragment {
  public static final String NEWLINE = "\r\n";
  @Inject Bus bus;
  @Bind(R.id.lyrics_recycler_view) RecyclerView recyclerView;

  @Override public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
  }

  @Override public void onViewCreated(View view, Bundle savedInstanceState) {
    super.onViewCreated(view, savedInstanceState);
    recyclerView.setHasFixedSize(true);
    RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(getActivity());
    recyclerView.setLayoutManager(layoutManager);
  }

  @Override public View onCreateView(LayoutInflater inflater, ViewGroup container,
      Bundle savedInstanceState) {
    final View view = inflater.inflate(R.layout.ui_fragment_lyrics, container, false);
    ButterKnife.bind(this, view);
    return view;
  }

  @Override public void onStart() {
    super.onStart();
    bus.register(this);
  }

  @Override public void onStop() {
    super.onStop();
    bus.unregister(this);
  }

  @Subscribe public void updateLyricsData(LyricsUpdated update) {
    final ArrayList<String> lyrics =
        new ArrayList<>(Arrays.asList(update.getLyrics().split(NEWLINE)));
    LyricsAdapter adapter = new LyricsAdapter(getActivity(), lyrics);
    recyclerView.setAdapter(adapter);
  }
}

File: app/src/main/java/com/kelsos/mbrc/ui/navigation/lyrics/LyricsFragment.kt
package com.kelsos.mbrc.ui.navigation.lyrics

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.constraintlayout.widget.Group
import androidx.core.view.isVisible
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.kelsos.mbrc.R
import kotterknife.bindView
import org.koin.android.ext.android.inject

class LyricsFragment : Fragment() {

  private val lyricsRecycler: RecyclerView by bindView(R.id.lyrics__lyrics_list)
  private val emptyView: Group by bindView(R.id.lyrics__empty_group)

  private val viewModel: LyricsViewModel by inject()
  private val lyricsAdapter: LyricsAdapter by lazy { LyricsAdapter() }

  private fun setupRecycler() {
    lyricsRecycler.apply {
      setHasFixedSize(true)
      layoutManager = LinearLayoutManager(requireContext())
      adapter = lyricsAdapter
    }
  }

  override fun onCreateView(
    inflater: LayoutInflater,
    container: ViewGroup?,
    savedInstanceState: Bundle?
  ): View? {
    return inflater.inflate(R.layout.fragment_lyrics, container, false)
  }

  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
    super.onViewCreated(view, savedInstanceState)
    setupRecycler()

    viewModel.lyricsLiveDataProvider.observe(this) { lyrics ->
      emptyView.isVisible = lyrics.isEmpty()
      lyricsAdapter.submitList(lyrics)
    }
  }
}